Source/events/EventConverters.ts

```diff
@@ -46,6 +46,21 @@ export class EventConverters {
         return uncommittedEvent;
     }
 
+    /**
+     * Creates an uncommitted embedding event (aka an event without an eventsourceid) from given parameters.
+     * @param {*} event Event content to constructor with.
+     * @param {EventType} eventType The event type.
+     * @param {boolean} isPublic Whether or not it is a public event
+     * @returns {UncommittedEvent} Constructed uncommitted event.
+     */
+    static getUncommittedEmbeddingEventFrom(event: any, eventType: EventType, isPublic: boolean): PbUncommittedEvent {
+        const uncommittedEvent = new PbUncommittedEvent();
+        uncommittedEvent.setArtifact(eventTypes.toProtobuf(eventType));
+        uncommittedEvent.setPublic(isPublic);
+        uncommittedEvent.setContent(JSON.stringify(event));
+        return uncommittedEvent;
```

Comment:
What about the event source id

Reply:
The Runtime is keeping track (and assigning)  of the event source id's, right? It will get these events without an event source id, set it to them and then try to handle them on the embedding. The contracts don't even have the event source id on them, as it's assigned automatically by the Runtime in the background. Or this has been my understanding of how this is going to work.

Reply:
After looking at this together it is the Runtime's responsibility to set the correct event source id. I'll resolve this